src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -4,61 +4,313 @@
+    hstring KeyBindingViewModel::CancelButtonName() const noexcept { return RS_(L"Actions_CancelButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"); }
+    hstring KeyBindingViewModel::AcceptButtonName() const noexcept { return RS_(L"Actions_AcceptButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"); }
+    hstring KeyBindingViewModel::DeleteButtonName() const noexcept { return RS_(L"Actions_DeleteButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"); }
+
+    bool KeyBindingViewModel::ShowEditButton() const noexcept
+    {
+        return (IsContainerFocused() || IsEditButtonFocused() || IsHovered() || IsAutomationPeerAttached()) && !IsInEditMode();
+    }
+
+    void KeyBindingViewModel::ToggleEditMode()
+    {
+        // toggle edit mode
+        IsInEditMode(!_IsInEditMode);
+        if (_IsInEditMode)
+        {
+            // if we're in edit mode,
+            // pre-populate the text box with the current keys
+            ProposedKeys(KeyChordText());
+        }
+    }
+
+    void KeyBindingViewModel::AttemptAcceptChanges()
+    {
+        AttemptAcceptChanges(_ProposedKeys);
+    }
+
+    void KeyBindingViewModel::AttemptAcceptChanges(hstring newKeyChordText)
+    {
+        auto args{ make_self<RebindKeysEventArgs>(_Keys, _Keys) };
+        try
+        {
+            // Attempt to convert the provided key chord text
+            const auto newKeyChord{ KeyChordSerialization::FromString(newKeyChordText) };
+            args->NewKeys(newKeyChord);
+            _RebindKeysRequestedHandlers(*this, *args);
+        }
+        catch (hresult_invalid_argument)
+        {
+            // Converting the text into a key chord failed
+            // TODO GH #6900:
+            //  This is tricky. I still haven't found a way to reference the
+            //  key chord text box. It's hidden behind the data template.
+            //  Ideally, some kind of notification would alert the user, but
+            //  to make it look nice, we need it to somehow target the text box.
+            //  Alternatively, we want a full key chord editor/listener.
+            //  If we implement that, we won't need this validation or error message.
+        }
+    }
+
     Actions::Actions()
     {
         InitializeComponent();
+    }
 
-        _filteredActions = winrt::single_threaded_observable_vector<Command>();
+    Automation::Peers::AutomationPeer Actions::OnCreateAutomationPeer()
+    {
+        for (const auto& kbdVM : _KeyBindingList)
+        {
+            // To create a more accessible experience, we want the "edit" buttons to _always_
+            // appear when a screen reader is attached. This ensures that the edit buttons are
+            // accessible via the UIA tree.
+            get_self<KeyBindingViewModel>(kbdVM)->IsAutomationPeerAttached(_AutomationPeerAttached);

```

Comment:
who sets `_AutomationPeerAttached`?

Reply:
WHOOPS! I originally had `_AutomationPeerAttached = true;` at the top of `OnCreateAutomationPeer`, but I removed it because my machine was _also_ always creating automation peers when I wasn't using any assistive tech. Added this back in.